app/assets/javascripts/discourse/app/initializers/welcome-topic-banner.js

```diff
@@ -0,0 +1,16 @@
+export default {
+  name: "welcome-topic-banner",
+  after: "message-bus",
+
+  initialize(container) {
+    const messageBus = container.lookup("service:message-bus");
+    if (!messageBus) {
+      return;
+    }
+
+    const site = container.lookup("service:site");
+    messageBus.subscribe("/site/welcome-topic-banner", function (disabled) {
```

Comment:
I don't think we need this subscription if `Site#show_welcome_topic_banner` is already false.

Reply:
Good idea, I've made this change.